sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -666,7 +666,12 @@ def _create_table_if_needed(self, table_reference, schema=None):
     logging.debug('Creating or getting table %s with schema %s.',
                   table_reference, schema)
 
-    table_schema = self.get_table_schema(schema)
+    if schema == SCHEMA_AUTODETECT:
```

Comment:
Are we assuming that this helper only gets called only in streaming inserts codepath? 
Perhaps we should strengthen the condition to something like: `if schema == SCHEMA_AUTODETECT and  method == STREAMING_INSERTS`, or move this check somewhere up the call stack to fail early?